### PR TITLE
Tag nat gateways

### PR DESCRIPTION
**Pull request by mikesplain**
Add tags to nat gateways

Fixes https://github.com/kubernetes/kops/issues/4604

### CODE REVIEW

**Comment by robinpercy** [comment]
Thanks @mikesplain, will give this a test shortly

File: pkg/model/network.go [hunk 1]
```diff
@@ -276,6 +277,7 @@ func (b *NetworkModelBuilder) Build(c *fi.ModelBuilderContext) error {
 				Subnet:               utilitySubnet,
 				ElasticIP:            eip,
 				AssociatedRouteTable: b.LinkToPrivateRouteTableInZone(zone),
+				Tags:                 tags,
```
Comment (chrislovecnm) [inline pkg/model/network.go]: Before I forget, I know @robinpercy will review, but @justinsb had to setup where we had to not just set the tags, but had to jump through a couple of hoops. See https://github.com/kubernetes/kops/pull/4572

I do not know if this PR actually needs that or not ...

**Comment by robinpercy** [comment]
Tested and works as expected 👍 

LGTM, barring comment from @justinsb on  https://github.com/kubernetes/kops/pull/4606#discussion_r173005912. 

From what I can see, we're currently using the tags on the associated subnets when looking up NGWs. So, there may be some new optimizations available, but I don't think this change makes anything worse.

/lgtm

**Comment by mikesplain** [comment]
Thanks for the review @robinpercy.  I think @chrislovecnm may be right, those changes @justinsb made should be easy enough to do, going to test them in a sec.

**Comment by mikesplain** [comment]
Testing this, I think it needs a few more changes.
/hold

**Comment by mikesplain** [comment]
Alrighty, sorry for the changes but based on that other PR, I think they're worth it. @robinpercy can you take a look again? Any comments @justinsb?

**Comment by robinpercy** [comment]
No problem @mikesplain the change makes sense 👍 
/lgtm

**Pull request merged**